uploader/api_commands.py

```diff
@@ -1,65 +1,62 @@
 
     # Create the current date folders
-    current_unprocessed_folder = os.path.join(app.config['UNPROCESSED_FOLDER'], current_date)
-    current_processed_folder = os.path.join(app.config['PROCESSED_FOLDER'], current_date)
+    current_unprocessed_folder = os.path.join(unprocessed_folder, current_date)
+    current_processed_folder = os.path.join(processed_folder, current_date)
     os.makedirs(current_unprocessed_folder, exist_ok=True)
     os.makedirs(current_processed_folder, exist_ok=True)
 
-    # Save the file
-    file = request.files['file']
-    # TODO: As we are planning to use auth, we should use the user id as a prefix for the filename
-    filename = secure_filename(file.filename)
+    filename = secure_filename(file.filename) if file.filename else None
+    if not filename:
+        return 'Invalid file'
+
     filename_without_ext = os.path.splitext(filename)[0]
     extension = os.path.splitext(filename)[1]
     filename_with_timestamp = f"{filename_without_ext}_{current_time}{extension}"
 
-    # TODO: We need to use the DB container to store the file url with status 'unprocessed'
+    # Save the file
     file.save(os.path.join(current_unprocessed_folder, filename_with_timestamp))
-
-    # TODO: We need to return here the file id to the user
-
-    # TODO: process the video in a separate function, We should use a task queue like Celery for this. this function is blocking and will not scale well
-
-    # Process the video
-    # TODO: We need to update the DB register to status 'processing'
-    unprocessed_video = VideoFileClip(
-        os.path.join(current_unprocessed_folder, filename_with_timestamp))
-
-    # Shorten the video if it's longer than 20 seconds
-    if unprocessed_video.duration > 20:
-        unprocessed_video = unprocessed_video.subclip(0, 20)
-
-    # Add the IDRL logo at the beginning and end of the video
-    idrl_logo = ImageClip(app.config["LOGO_FILE"], duration=1)
-    processed_video = concatenate_videoclips([idrl_logo, unprocessed_video, idrl_logo])
-
-    # Change the aspect ratio to 16:9
-    processed_video = processed_video.resize(height=720)  # Resize height to 720p
-    processed_video = processed_video.crop(x_center=processed_video.w / 2, y_center=processed_video.h / 2, width=1280,
-                                 height=720)  # Crop to 16:9
-
-    # Save the final video
-    # TODO: We need to update the DB register to status 'processed' and update the file url with the processed file
-    processed_video.write_videofile(
-        os.path.join(current_processed_folder, 'processed_' + filename_with_timestamp))
+    file.close()
+
+    args = [
+        filename_with_timestamp,
+        current_unprocessed_folder,
+        current_processed_folder,
+        logo_file
+    ]
+    print(args)
+
+    # Call celery
+    procesar_video.apply_async(args=args, queue='batch_videos')
+    # TODO: We need to use the DB container to store the file url with status 'unprocessed'
```

Comment:
Creo que para esta tarea necesitamos que de alguna manera:

1. el batch tenga acceso a la base de datos para modificar El estado del video .
2. Que el uploader tenga otro proceso escuchando el la respuesta del task de celery. para que actualice el estado final.

Reply:
Should be fixed now